dxp/liferay_global_event_handler.md

```diff
@@ -0,0 +1,213 @@
+# Liferay’s Global Event Handler
+
+Liferay’s event handler helps the developers for creating/listening/firing/removing custom events that can be listened in whole DXP which is rely on the YUI3 event target implementation. See the implementation [here](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L120).
+
+### First Considerations
+
+First, just consider using the global event handler if necessary. Try to use the default event listener/[callback pattern(if React)](https://reactjs.org/docs/lifting-state-up.html#lifting-state-up) when working. Just use the global event handler when trying to communicate between widgets or not accessible contexts(e.g iframes).
+
+### List of Available Functions
+
+Here is a list of available functions from Liferay’s global variable:
+
+`Liferay.on`
+`Liferay.once`
+`Liferay.onceAfter`
+`Liferay.after`
+`Liferay.before`
+`Liferay.detach`
+`Liferay.detachAll`
+`Liferay.publish`
+`Liferay.fire`
+
+### `Liferay.on`
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L217):
+
+This function subscribes a callback function to a custom event fired by this object or from an object that bubbles its events to this object.
+
+Callback functions for events published with `emitFacade = true` will receive an `EventFacade` as the first argument (typically named “e”). These callbacks can then call `e.preventDefault()` to disable the behaviour published to that event’s `defaultFn` . See the `EventFacade` API for all available properties and methods. Subscribers to non- `emitFacade` events will receive the arguments passed to `fire()` after the event name.
+
+To subscribe to multiple events at once, pass an object as the first argument, where the key:value pairs correspond to the eventName:callback, or pass an array of event names as the first argument to subscribe to all listed events with the same callback.
```

Comment:
I don't think anybody is doing this in liferay-portal, so I wouldn't document it:

```
git grep 'Liferay.on\(\{'
```

... returns no hits.

Reply:
Now I got it, people weren't using the configs parameter

Reply:
Yeah, I was just talking about the functionality described on line 31; which nobody is using (unless it's hidden very deeply... Prettier should always keep the curly on the same line after `Liferay.on(`, which is why I did the grep like that).